fb: cfb: add API to set font kerning

5frank opened the pull request:
Font kerning was used but it was not possible to change it. Also, font_idx was set to zero twice in init.

Signed-off-by: Simon Frank <simon.frank@lohmega.com>

jfischer-no reviewed (changes requested):
Please change commit title to `fb: cfb: add API to set font kerning `, thanks

soburi commented on subsys/fb/cfb.c:
```diff
@@ -343,7 +351,6 @@ int cfb_framebuffer_init(const struct device *dev)
 	fb->pixel_format = cfg.current_pixel_format;
 	fb->screen_info = cfg.screen_info;
 	fb->buf = NULL;
-	fb->font_idx = 0U;
```
Is it need to remove?

5frank replied:
yes. same line twice. please see commit message.

soburi replied:
Sorry, I missed the comments.
It's no problem.

soburi commented on include/zephyr/display/cfb.h:
```diff
@@ -153,6 +153,16 @@ int cfb_get_display_parameter(const struct device *dev,
+ *
+ * @return 0 on success, negative value otherwise
+ */
+int cfb_set_kerning(const struct device *dev, int8_t kerning);
```
Could you please add command to shell that invoke the API?

5frank replied:
added command `set_kerning` to cfb_shell.c

soburi commented on subsys/fb/cfb.c:
```diff
@@ -316,6 +316,14 @@ int cfb_get_font_size(const struct device *dev, uint8_t idx, uint8_t *width,
+	return 0;
+}
+
+
```
Blank line is usually 1-line

soburi commented on subsys/fb/cfb_shell.c:
```diff
@@ -235,6 +235,33 @@ static int cmd_set_font(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+	char *ep = NULL;
+	long int kerning;
```
Please Fix CI check.

soburi commented on subsys/fb/cfb_shell.c:
```diff
@@ -235,6 +235,33 @@ static int cmd_set_font(const struct shell *shell, size_t argc, char *argv[])
 	return err;
 }
 
+static int cmd_set_kerning(const struct shell *shell, size_t argc, char *argv[])
```
Could you please separate commits about `cfb_shell.c` modification?

Commit:
fb: cfb: add API to set font kerning

Font kerning was used but it was not possible to change it.
Also, font_idx was set to zero twice in init.

Signed-off-by: Simon Frank <simon.frank@lohmega.com>

Commit:
fb: cfb: cfb_shell command to set font kerning

Using `struct shell *shell` would be more consistent but triggers CI
check on MISRA rule 5.7 "A Tag name shall be a unique identifier"

Signed-off-by: Simon Frank <simon.frank@lohmega.com>

soburi commented:
@5frank 
LGTM. Thank you for contribution!

@jfischer-no 
Please take a look.
